pkg/scripts_test/install_unix_test.go

```diff
@@ -7,21 +7,58 @@ import (
 )
 
 func TestInstallScript(t *testing.T) {
+	notInstalledChecks := []checkFunc{
+		checkBinaryNotCreated,
+		checkConfigNotCreated,
+		checkUserConfigNotCreated,
+		checkUserNotExists,
+		checkGroupNotExists,
+		checkUserConfigNotCreated,
+		checkSystemdConfigNotCreated,
+		checkDeprecatedSystemdConfigNotCreated,
+	}
+
 	for _, spec := range []testSpec{
```

Comment:
Is there a simple way to avoid all this duplication? For example, take the testspec slice, run all the tests as they are before this change, and then run them again manually enabling native packaging?

Reply:
The main issue is that the PreActions, PreChecks, PostChecks, etc. differ between binary & package installs in most if not all of the specs. Reducing duplication may not be worth the time investment as we're planning on removing binary installation support from the install script.

Reply:
Intuitively, the install script should give the same results whether using native packaging or not, at least with systemd enabled. What is the source of the differences?

Reply:
The majority of the difference is in the pre-actions & pre-checks.

I avoided using much of the config mocking used in the tests to prevent package installation conflicts around existing config files. This isn't a problem when installing under an interactive tty as the installer can prompt how to handle conflicts. To work around this issue I wrote a helper to install a previous version of the package (defined by the const `previousVersion`) in some of the preActions of specs. This should work with binary installations but versions differ between binaries and packages to allow for out-of-band changes to packaging code.

The binary install uses the wrong path for the systemd unit file which has been fixed in the packages. I had assumed, perhaps incorrectly, that we wouldn't want to change paths of any files for binary installations. We can change the path to match the packages. We'll need logic to handle the removal or migration of the unit file at the old path.

Reply:
Allright, I'm fine with that. Can we then move the native test cases to a separate test? There's a lot of them and I'd like to introduce some basic organization here, to make changes easier in the future.